crates/router/src/core/payments/helpers.rs

```diff
@@ -987,14 +987,16 @@ where
         match schedule_time {
             Some(stime) => {
                 if !requeue {
-                    // scheduler_metrics::TASKS_ADDED_COUNT.add(&metrics::CONTEXT, 1, &[]); // Metrics
+                    // Increment the count of added tasks every time a payment has been confirmed or PSync has been called
```

Comment:
Task addition is not limited to a payment being confirmed or a payment sync occurring though.